### ISSUE TITLE
Update ACF blocks via block API dosen't work as expected

### ISSUE THREAD

**Issue by CreativeDive** [AdvancedCustomFields/acf#817]
Hey @lgladdy,

lately I've been learning more about the WordPress Block Editor API. There are many useful functions that we can use to interact with blocks. I noticed that using the updateBlockAttributes() function applied to ACF blocks does not work as expected.

```javascript
wp.data.dispatch('core/block-editor').updateBlockAttributes(clientId, {
    data: { ... },
});
```

What works is that after updating the ACF block attributes in the above way, the block is updated with the newly set attributes. It seems `render_block_preview` gets the job done after swapping the block attributes.

But the BlockForm in the sidebar contains the old settings. That means, if I change a field after updating the block in the above way, the updated block reverts to the original state.

The behavior is understandable as the BlockForm was not updated when the Block was updated via the API. Unfortunately, this causes updating the block not to work that way.

Ultimately, only the BlockForm would have to be updated. Is it possible for ACF to update the BlockForm as well?

**Comment by lgladdy** [comment]
Hey @CreativeDive.

Updating ACF attributes this way isn't supported yet as this won't be able to update all our internal logic and components.

Pinging @polevaultweb to add to the feature request list :)

**Comment by CreativeDive** [comment]
@lgladdy Yes that is what I thought. It would be a great extension. I have created a function that allows to copy and paste defined style attributes of an ACF block to another ACF block via the block settings. That works great. It works the same as between core blocks when styles are copied and pasted into another block. The only thing missing is updating the BlockForm after updating the block attributes.

**Comment by CreativeDive** [comment]
@polevaultweb Any chance of getting this issue higher on the todo list? I think if ACF meets the basic requirement to copy and paste styles it can be very helpful for many users. In order to meet the requirements for this, only the above-mentioned issue would have to be solved.

Currently you have to adjust everything again for each ACF block, if you can copy and paste the styles that would help a lot.

If you have 10 blocks that look the same, you have to change it 10 times if something needs to be changed.

**Comment by nick6352683** [comment]
+1

**Comment by CreativeDive** [comment]
I wish there was a public voting list. You don't get any feedback and you don't know if and when certain features will be implemented.
